controller.go

```diff
@@ -238,87 +209,52 @@ func (c *controller) NewNetwork(networkType, name string, options ...NetworkOpti
 
 	// Construct the network object
 	network := &network{
-		name:      name,
-		id:        types.UUID(stringid.GenerateRandomID()),
-		ctrlr:     c,
-		driver:    d,
-		endpoints: endpointTable{},
+		name:        name,
+		networkType: networkType,
+		id:          types.UUID(stringid.GenerateRandomID()),
+		ctrlr:       c,
+		endpoints:   endpointTable{},
 	}
 
 	network.processOptions(options...)
-	if err := c.addNetworkToStore(network); err != nil {
+
+	if err := c.addNetwork(network); err != nil {
 		return nil, err
 	}
-	// Create the network
-	if err := d.CreateNetwork(network.id, network.generic); err != nil {
+
+	if err := c.addNetworkToStore(network); err != nil {
```

Comment:
Good point. we currently dont have the distributed error handler that we discussed and hence addressing this use-case alone is not sufficient to handle the global consistent view between the local cache, programmed state and the distributed store states. the distributed error handler will be addressed as a next step in this series of distributed store commits.